packages/picasso-charts/src/BarChart/BarChart.tsx

```
@@ -106,8 +113,19 @@ const BarChart = <K extends string>({
   showBarLabel,
   isAnimationActive,
   layout,
+  showEveryNthTickOnXAxis = 1,
+  showEveryNthTickOnYAxis = 1,
   ...rest
 }: Props<K>) => {
+  if (
+    !isNonNegativeInteger(showEveryNthTickOnXAxis) ||
+    !isNonNegativeInteger(showEveryNthTickOnYAxis)
+  ) {
+    throw new Error(
+      'Non-negative integer values are expected for "showEveryNthTick*" properties'
+    )
+  }
+
```

Hmm, don't we except `0 | 1 | 2 | 3` instead of `number`? Then this check would not be needed.
Even if it is not possible, I don't think we should react with error and app crash. If something unexpected passed I believe we can simply not show ticks, or use default value. This would serve users and will not flood us with ERs :)

---

@OleksandrNechai This is done to give more options to end-users after https://github.com/toptal/picasso/pull/3550#discussion_r1191184471 proposal, so it can no longer be easily typed to specific integers in TypeScript

---

> If something unexpected passed I believe we can simply not show ticks, or use the default value.

I think if the error is not thrown, then it needs to be handled gracefully but the warning still should be shown which complicates the behavior (ERs will be assigned to those who misused the components)

---

>This is done to give more options to end-users after https://github.com/toptal/picasso/pull/3550#discussion_r1191184471 proposal

I see. Picasso's philosophy has always been `less is more. We are limiting APIs on purpose to increase consistency in our apps. Also, it is much easier to change 0 | 1 | 2 | 3 --> number than to change it back. 

>I think if the error is not thrown, then it needs to be handled gracefully 

Yes, there is an idea to fail early. It is easier to debug when we do not swallow the error in places where it happens. But we are doing a highly reusable library. To me, Postel's law feels more appropriate https://devopedia.org/postel-s-law 
>Be liberal in what you accept, and conservative in what you send.

crashing seems too much in this situation

---

> We are limiting APIs on purpose to increase consistency in our apps.

Can you please specify why using enums is considered to be more consistent in this case? Picasso components API's do not really force it and I can see that `someProp?: number` is very common in Picasso APIs

> crashing seems too much in this situation

Thanks for the link! However, I do think that in this case it is still suitable, as there is a high chance that either exception will be thrown when testing (if an engineer sets the wrong property value) or it will be caught on review. This is basic data validation of the input, so silently substituting user's input will make things more obscure from consumer point of view which is not a good approach.

---

Clarifying the Nth ticks display question in https://toptal-core.slack.com/archives/GG3F4AS4T/p1683882228257659

---

Decision https://toptal-core.slack.com/archives/GG3F4AS4T/p1683884552949139